admin/console/dangerzone-for-experts.rst

```diff
@@ -29,76 +29,85 @@ Deleting Customers
 
    As such, the examples below will delete not only the specified customers, but **all tickets associated with them**, as well.
 
-.. code-block:: ruby
+Step 1: Select customers by email address
+   .. code-block:: ruby
+
+      >> customers = User.where(email: %w[customer@example.com customer@example.org])
+
+      >> customers = customers.joins(roles: :permissions)
+                              .where(roles: { active: true })
+                              .where(permissions: { name: 'ticket.customer', active: true })
+                              .where.not(id: 1)
```

Comment:
When copying and executing this code part:

```
customers = User.where(email: %w[example1@domain.de example2@domain.de])
customers = customers.joins(roles: :permissions)
                              .where(roles: { active: true })
                              .where(permissions: { name: 'ticket.customer', active: true })
                              .where.not(id: 1)
```

I receive the following error:

```
2.5.5 :003 >                               .where(roles: { active: true })
Traceback (most recent call last):
SyntaxError ((irb):3: syntax error, unexpected '.')
                              .where(roles: { active: true })
                              ^
2.5.5 :004 >                               .where(permissions: { name: 'ticket.customer', active: true })
Traceback (most recent call last):
SyntaxError ((irb):4: syntax error, unexpected '.')
                              .where(permissions: { name: 't...
                              ^
2.5.5 :005 >                               .where.not(id: 1)
puts customers.map do |user|
Traceback (most recent call last):
SyntaxError ((irb):5: syntax error, unexpected '.')
                              .where.not(id: 1)
                              ^
2.5.5 :006 >
```

This is due to the spacings living inside the code block.
I code solve this with using ` \` on each line end like so:

```
customers = User.where(email: %w[example1@domain.de example2@domain.de])
customers = customers.joins(roles: :permissions) \
                              .where(roles: { active: true }) \
                              .where(permissions: { name: 'ticket.customer', active: true }) \
                              .where.not(id: 1)
```

However: I do not know if this breaks anything or brings other kind of issues.
Could you please check this? :)

Reply:
Sorry for the delay. Yes, this is a problem with the way I wrote the code sample.

Ruby allows you to chain method calls like this in a script, but not in the console. In a script, the interpreter looks ahead, sees the `.where` on the next line, and includes that as part of the first line. The console can't do that, and so when you get to the end of `customers.joins(...)` then it just stops there and evaluates what it gets.

The backslashes will fix this without introducing any other issues. Thanks for looking out.

Reply:
In the end, I discovered a portability/comatibility issue with backslashes, so instead I joined the entire `.joins` / `.where` query onto a single line.

Specifically, the Rails console is built on `irb`, but there's an irb alternative out there called pry (which some folks on the development team use), and pry doesn't handle lines of Ruby that start with a dot.